Reactor/Reactor/ReactorConfiguration.swift

```diff
@@ -6,33 +6,63 @@
 //  Copyright © 2016 Mail Online. All rights reserved.
 //
 
-/// Configuration object to customize the Reactor's behaviour
-public struct ReactorConfiguration {
-
-    /// If persistence should be used.
-    /// `true` by default.
-    public var usingPersistence: Bool = true
-    /// If reachability should be used.
-    /// `true` by default.
-    public var shouldCheckReachability: Bool = true
+public protocol CoreConfigurable {
     /// If the entire flow should fail, when `saveToPersistenceFlow` fails.
     /// `true` by default.
-    public var flowShouldFailWhenSaveToPersistenceFails: Bool = true
+    var shouldFailWhenSaveToPersistenceFails: Bool { get }
     /// If the `saveToPersistenceFlow`, should be part of the flow.
     /// Should be `false` when the flow shouldn't
-    /// wait for `saveToPersistenceFlow` to finish (for example it takes 
+    /// wait for `saveToPersistenceFlow` to finish (for example it takes
     /// a long time).
-    /// Note: if you set it as `false` and it fails, the failure will be 
+    /// Note: if you set it as `false` and it fails, the failure will be
     /// lost, because it's not part of the flow, but injected instead .
     /// `true` by default.
+    var shouldWaitForSaveToPersistence: Bool { get }
+}
+
+extension CoreConfiguration: CoreConfigurable {}
+
+/// Configuration object to customize the Reactor's behaviour
+public struct CoreConfiguration {
+    
+    public var shouldFailWhenSaveToPersistenceFails: Bool = true
     public var shouldWaitForSaveToPersistence: Bool = true
+}
+
+public protocol FlowConfigurable {
+    /// If persistence should be used.
+    /// If it should a path needs to be provided. 
+    var usingPersistence: ShouldUsePersistence { get }
+    /// If reachability should be used.
+    /// `true` by default.
+    var shouldCheckReachability: Bool { get }
     /// If the parser should be strick or prune the bad objects.
```

Comment:
See [this comment](https://github.com/MailOnline/Reactor/pull/17/files#r64933908)